### PR TITLE
Connector uses unsupported API (#47)

### PR DESCRIPTION
**Pull request by PhilMFischer**
- Fixed compile issues due to changes in java-gitlab-api
- Prepared for Including local Snapshot build of java-gitlab-api 4.0.1 with fixes of https://github.com/timols/java-gitlab-api/pull/284
- Added snapshots to gitignore file

- Replaced external libraries as needed by java gitlab api 4.0.x
- Added jackson-core 2.5.3
- Added jackson-databind 2.5.3
- Added jackson-annotations 2.5.3
- Added commons-io 2.4

- Updated Licensing information to show about and added third party licenses as much as they are available. Now following eclipse style handling the licenses in plugins.
- Updated plugin license information for java-gitlab-api

### CODE REVIEW

File: de.weingardt.mylyn.gitlab.core/about.html [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
+<title>About</title>
+</head>
+<body lang="EN-US">
+<h2>About This Content</h2>
+ 
+<p>July, 2016</p>	
+<h3>License</h3>
+
+<p> Unless otherwise indicated below, all content of this plugin &quot;Content&quot; is provided to you under the terms
```
Comment (PhilMFischer) [inline de.weingardt.mylyn.gitlab.core/about.html]: @pweingardt usually if you look into the about.html of eclipse plugins. The author is emntiones as well in a way such as "The content of this plugin is made available by pweingart". Was not sure if you would like it or not. Please check if you like it or not.

**Comment by samrocketman** [comment]
IMO the license bit probably should have been a separate PR.  You've mixed a couple issues in one PR so it became pretty big (24 files is not a small amount).

It's not clear if the license files were copied from the repository license or elsewhere at a glance.

**Review by maba001 — APPROVED** [review]
I came to the exact same change results (apart from the doc / license enhancements). In my opinion this looks okay to be merged.

**Comment by vincowl** [comment]
@PhilMFischer How would it be possible to test it locally ? I mean deploy it in a Photon Eclipse installation for instance. Sadly, I don't know much about Eclipse plugins...

**Comment by pweingardt** [comment]
I will look into this pull request on the weekend.

**Comment by fogs** [comment]
Would it help to integrate this PR if we split it into two separate ones; one for API v4 and one for the licensing? Or get you a beer or a coffee? ;)

**Pull request merged**